backend/sirius-web-diagrams-layout/src/main/java/org/eclipse/sirius/web/diagrams/layout/incremental/IncrementalLayoutEngine.java

```diff
@@ -83,12 +83,20 @@ public void layout(Optional<IDiagramEvent> optionalDiagramElementEvent, DiagramL
 
         // finally we recompute the edges that needs to
         for (EdgeLayoutData edge : diagram.getEdges()) {
-            if (this.hasChanged(edge.getSource()) || this.hasChanged(edge.getTarget())) {
+            if (this.hasChanged(edge.getSource()) || this.hasChanged(edge.getTarget()) || !this.isLabelPositioned(edge)) {
                 this.layoutEdge(edge);
             }
         }
     }
 
+    private boolean isLabelPositioned(EdgeLayoutData edge) {
+        if (edge.getCenterLabel() != null) {
```

Comment:
Any reason to only check for the center label and not the begin and end ones?

Reply:
I was wondering the same thing :)

Reply:
I've talked with @gcoutable and since only the center label is properly layout for now, he is only considering this one. I'll rebase and merge it since it improves things.